backend/server.js

```diff
@@ -961,10 +955,37 @@ app.get("/qInstances/:qiid/:filename", function(req, res) {
                 if (!_(info).has("clientFiles")) {
                     return sendError(res, 500, "Question does not have clientFiles, qid: " + fileInfo.qid);
                 }
-                if (!_(info.clientFiles).contains(fileInfo.filename)) {
-                    return sendError(res, 404, "Access denied to '" + fileInfo.filename + "' for qid: " + fileInfo.qid);
+
+                if (_(info).has("clientTemplates") && _(info.clientTemplates).contains(fileInfo.filename)) {
+                    // File is template which needs to be rendered
+                    fs.readFile(path.join(fileInfo.root, fileInfo.filePath), function(err, data) {
```

Comment:
Is there a risk here that `fileInfo.filePath` might be something like `../../../etc/ssh_host_key`? With the `res.sendFile()` call for non-templated files we avoid this risk by using the `root` option which blocks access to parent directories, but I'm a bit worried about `path.join()`.